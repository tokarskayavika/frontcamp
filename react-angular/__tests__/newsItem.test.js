import React from 'react';
import Enzyme from 'enzyme';
import NewsItem from '../scripts/ui/posts/NewsItem';
import Adapter from 'enzyme-adapter-react-16';
import expectFrom from 'expect';
import enzymify from 'expect-enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({
    adapter: new Adapter()
});

expectFrom.extend(enzymify());

describe('News Item Component', () => {
    let item;
    let props;

    beforeEach(() => {
        props = {
            author: '',
            description: '',
            title: ''
        };
        item = Enzyme.shallow(<NewsItem data={props} id={'0'} />);
    });

    it('component should contain title, description and author fields', () => {
        expectFrom(item).toContain('.title');
        expectFrom(item).toContain('.description');
        expectFrom(item).toContain('.author');
    });

    it('newsItem renders properly', () => {
        const item = renderer
            .create(<NewsItem data={props} id={'0'} />)
            .toJSON();
        expect(item).toMatchSnapshot();
    });
});