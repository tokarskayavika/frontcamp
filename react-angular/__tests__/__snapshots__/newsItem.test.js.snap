// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Item Component newsItem renders properly 1`] = `
<li
  className="news-item"
  data-id="0"
>
  <h3
    className="title"
  >
    
  </h3>
  <div
    className="feed-info"
  >
    <p
      className="author"
    >
      
    </p>
    <p
      className="description"
    >
      
    </p>
  </div>
</li>
`;
